### PR TITLE
feat: Add new HOWTO on creating vGPU servers

### PR DESCRIPTION
**Pull request by colder-is-better**
We demonstrate how to create servers with virtual CPU support, using the OpenStack CLI utility.

### CODE REVIEW

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 1]
```diff
@@ -0,0 +1,210 @@
+# Creating new servers with virtual GPU support
+
+In {{brand}}, you can instantiate virtual machines --- or simply
+_servers_ --- with virtual GPU (vGPU) support. For that, you may use the
+{{gui}} or the OpenStack CLI. Following both approaches, let us
+demonstrate the creation of a new server with vGPU support.
+
+## Prerequisites
+
+Although not necessary, it certainly helps a lot to have some
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:10]:
```suggestion
Although not strictly necessary, it is certainly helpful to have some
```

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+familiarity with [creating servers in
+{{brand}}](/howto/openstack/nova/new-server). Additionally, if you
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:12]: Don't use absolute links, and do make use of Mkdocs' magic linking ability referencing source Markdown files, like so:

```suggestion
{{brand}}](new-server.md). Additionally, if you
```

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+prefer to work with the OpenStack CLI, then make sure to properly
+[enable it first](/howto/getting-started/enable-openstack-cli).
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:14]: Here, too:
```suggestion
[enable it first](../../getting-started/enable-openstack-cli.md).
```

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+    {{brand}} account.
+
+    Create your new vGPU server, working almost exactly as you would when
+    creating a "normal" server. There are two steps that you should pay
+    special attention to, though.
+        
+    First, the image for the new server should be the `Ubuntu 22.04 NVGRID
+    x86_64`; that is your only option for a server with vGPU support.
+        
+    ![Select image with vGPU support](assets/new-vgpu-server/shot-01.png)
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:30]: The screenshot isn't included in the PR (this is more of a note-to-self, as I presume that this is why the PR is marked as a draft).

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+        
+    Then, while selecting a flavor for your new vGPU server, please keep in
+    mind that you should only choose one with a name prefixed by "`g.`". For
+    instance, flavor `g.8c32gb50` is suitable for a vGPU server, but flavor
+    `b.8c32gb` is not. Failing to select an appropriate flavor leads to a
+    server with no virtual GPU support, even when it is based on the `Ubuntu
+    22.04 NVGRID x86_64` image.
+        
+    ![Select proper flavor for a vGPU server](assets/new-vgpu-server/shot-02.png)
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:39]: Again, note-to-self: screenshot missing.

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 6]
```diff
@@ -0,0 +1,210 @@
...
+=== "{{gui}}"
+    Fire up your favorite web browser, navigate to the
+    [{{gui}}](https://{{gui_domain}}) start page, and log into your
+    {{brand}} account.
+
+    Create your new vGPU server, working almost exactly as you would when
+    creating a "normal" server. There are two steps that you should pay
+    special attention to, though.
+        
+    First, the image for the new server should be the `Ubuntu 22.04 NVGRID
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:27]: One "the" sounds superfluous here:

```suggestion
    First, the image for the new server should be `Ubuntu 22.04 NVGRID
```

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 7]
```diff
@@ -0,0 +1,210 @@
...
+00:03.0 Ethernet controller: Red Hat, Inc. Virtio network device
+00:04.0 SCSI storage controller: Red Hat, Inc. Virtio SCSI
+00:05.0 Communication controller: Red Hat, Inc. Virtio console
+00:06.0 3D controller: NVIDIA Corporation GA102GL [A10] (rev a1)
+00:07.0 Unclassified device [00ff]: Red Hat, Inc. Virtio memory balloon
+00:08.0 Unclassified device [00ff]: Red Hat, Inc. Virtio RNG
+```
+
+Notice the third-to-last line, which says `3D controller: NVIDIA
+Corporation`; that's good! To get much more information regarding the
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:178]: Since we can't always assume for certain that it is indeed the third-to-last line, I'd suggest to rephrase this as:

```suggestion
Notice the line that says `3D controller: NVIDIA
Corporation`; that's good! To get much more information regarding the
```

**Review by fghaas — COMMENTED** [review]
This looks quite good to me; I just have a few small suggestions.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 1 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:10]: Ah, yes.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 2 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:12]: Done (and yes, that's magical).

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 3 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:14]: Yes.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 4 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:30]: Yes, I expect to have two screenshots exactly, and I prefer to create those as soon as the `g`-flavors are visible in CCMP.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 5 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:39]: Again, those invisible-to-CCMP `g`-flavors...

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 6 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:27]: Aha, I guess it is.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 7 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:178]: Much better, yes. (And, yes, I tend to agree a lot today).

**Review by fghaas — CHANGES_REQUESTED** [review]
Rebase on current `main`, please.

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 8]
```diff
@@ -0,0 +1,171 @@
...
+familiarity with [creating servers in {{brand}}](new-server.md),
+especially with the OpenStack CLI.
+
+## Instantiating a vGPU server
+
+Before you begin, please make sure to source [your `openstack` RC file
+first](../../getting-started/enable-openstack-cli.md).
+
+As we discussed in the [guide on creating
+servers](/howto/openstack/nova/new-server), an `openstack` command for
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:19]: Drop the absolute link; it's unnecessary.

```suggestion
servers](new-server.md), an `openstack` command for
```

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 9]
```diff
@@ -0,0 +1,171 @@
...
+    --boot-from-volume $VOL_SIZE \
+    --network $NETWORK_NAME \
+    --security-group $SEC_GROUP_NAME \
+    --key-name $KEY_NAME \
+    --wait \
+    $SERVER_NAME
+```
+
+For servers with vGPU support, the only acceptable value for variable
+`IMAGE_NAME` is `Ubuntu 22.04 NVGRID x86_64`. Additionally, the value
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:35]: I would suggest you use non-breaking spaces in the image name here:
```suggestion
`IMAGE_NAME` is `Ubuntu 22.04 NVGRID x86_64`. Additionally, the value
```
(I guess this doesn't look to clear in the review web UI, hence the explanation.)

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 10]
```diff
@@ -0,0 +1,171 @@
...
+for variable `FLAVOR_ID` should be the ID of a flavor prefixed by
+"`g.`". To see all such flavors, along with the corresponding IDs, you
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:37]: Cross-reference please, and also you can drop the double quotes and the extra full-stop:
```suggestion
for variable `FLAVOR_ID` should be the ID of a flavor [prefixed by
`g`](../../../reference/flavors/index.md#compute-tiers). 
To see all such flavors, along with the corresponding IDs, you
```
But a different thought: why are we referring to flavors by their UUID here, when we implore users to refer to images by name and never use UUIDs? Why don't we suggest to use flavor names?

**Review by fghaas — COMMENTED** [review]
A few more thoughts inline.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 8 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:19]: Missed that one -- thanks!

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 9 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:35]: Sorry, but I cannot type a non-breaking space in Mac OS :/ Pressing the Option key together with the Space key is supposed to do the trick, but that doesn't seem to work.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 10 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md]: Uhm, yes, names for flavors -- just like names for images. I have adapted the prose accordingly.

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 11]
```diff
@@ -0,0 +1,170 @@
...
+Although not strictly necessary, it is certainly helpful to have some
+familiarity with [creating servers in {{brand}}](new-server.md),
+especially with the OpenStack CLI.
+
+## Instantiating a vGPU server
+
+Before you begin, please make sure to source [your `openstack` RC file
+first](../../getting-started/enable-openstack-cli.md).
+
+As we discussed in the [guide on creating servers](new-server.md), an
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:18]: This howto guide is not a part of a sequential series, so we cannot assume as certainty that the reader has read the other guide. Thus, this feels more natural to me:
```suggestion
As we discuss in the [guide on creating servers](new-server.md), an
```

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+    --boot-from-volume $VOL_SIZE \
+    --network $NETWORK_NAME \
+    --security-group $SEC_GROUP_NAME \
+    --key-name $KEY_NAME \
+    --wait \
+    $SERVER_NAME
+```
+
+For servers with vGPU support, the only acceptable value for variable
+`IMAGE_NAME` is `Ubuntu 22.04 NVGRID x86_64`. Additionally, the value
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:34]: Either:
```suggestion
For servers with vGPU support, the only acceptable value for the
`IMAGE_NAME` variable is `Ubuntu 22.04 NVGRID x86_64`. Additionally, the value
```
or (shorter, and hence preferred):
```suggestion
For servers with vGPU support, the only acceptable value for
`IMAGE_NAME` is `Ubuntu 22.04 NVGRID x86_64`. Additionally, the value
```

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 13]
```diff
@@ -0,0 +1,170 @@
...
+for variable `FLAVOR_NAME` should be a flavor name [prefixed by
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:35]: Idem.

File: docs/howto/openstack/nova/new-vgpu-server.md [hunk 14]
```diff
@@ -0,0 +1,170 @@
...
+
+| g.8c48gb50                 |
+| g.24c120gb50               |
+| g.48c240gb                 |
+| g.12c48gb50                |
+| g.16c64gb50                |
+| g.12c64gb50                |
+| g.16c84gb50                |
+| g.8c32gb50                 |
+```
```
Comment (fghaas) [inline docs/howto/openstack/nova/new-vgpu-server.md:50]: If you add `-f value` to your `openstack` invocation, you can rid your screen dump of the ugly `|` characters. :)

**Review by fghaas — COMMENTED** [review]
Looks good now, I have a few remaining micro-nits.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 11 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:18]: It is more natural (and correct), yes.

File: docs/howto/openstack/nova/new-vgpu-server.md (hunk 12 above)
Comment (colder-is-better) [reply docs/howto/openstack/nova/new-vgpu-server.md:34]: The shorter!

**Review by fghaas — COMMENTED** [review]
This looks good to me now. I will shortly do a force-push to fix the non-breaking spaces, and then I'll approve and merge.

**Review by fghaas — APPROVED** [review]

**Comment by colder-is-better** [comment]
> This looks good to me now. I will shortly do a force-push to fix the non-breaking spaces, and then I'll approve and merge.

Very nice. Out of curiosity, how do you type non-breaking spaces? Which application and what key combination does the trick?

**Comment by fghaas** [comment]
I have my Compose key mapped to Caps Lock using GNOME Tweaks, so a non-breaking space amounts to typing the sequence `CapsLock-Space-Space` for me. Just like, say, å works via `CapsLock-o-a`.

**Comment by colder-is-better** [comment]
> I have my Compose key mapped to Caps Lock using GNOME Tweaks, so a non-breaking space amounts to typing the sequence `CapsLock-Space-Space` for me. Just like, say, å works via `CapsLock-o-a`.

Nice. I will somehow come up with something similar for OS X.
